src/theory/quantifiers/entailment_check.cpp

```diff
@@ -297,41 +297,31 @@ bool EntailmentCheck::isEntailed2(TNode n,
   Trace("term-db-entail") << "Check entailed : " << n << ", pol = " << pol
                           << std::endl;
   Assert(n.getType().isBoolean());
-  if (n.getKind() == EQUAL && !n[0].getType().isBoolean())
+  Kind k = n.getKind();
+  if (k == EQUAL && !n[0].getType().isBoolean())
   {
-    TNode n1 = getEntailedTerm2(n[0], subs, subsRep);
+    TNode n1 = n[0].isConst() ? n[0] : getEntailedTerm2(n[0], subs, subsRep);
     if (!n1.isNull())
     {
-      TNode n2 = getEntailedTerm2(n[1], subs, subsRep);
+      TNode n2 = n[1].isConst() ? n[1] : getEntailedTerm2(n[1], subs, subsRep);
       if (!n2.isNull())
       {
-        if (n1 == n2)
+        if (pol)
         {
-          return pol;
-        }
-        else
-        {
-          Assert(d_qstate.hasTerm(n1));
-          Assert(d_qstate.hasTerm(n2));
-          if (pol)
-          {
-            return d_qstate.areEqual(n1, n2);
-          }
-          else
-          {
-            return d_qstate.areDisequal(n1, n2);
-          }
+          // must check for equality here
+          return d_qstate.areEqual(n1, n2);
         }
+        return d_qstate.areDisequal(n1, n2);
```

Comment:
```suggestion
        // must check for equality here
        return pol? d_qstate.areEqual(n1, n2) : d_qstate.areDisequal(n1, n2);
```
and deleting the `if` above (github did not let me easily refer to it)

Reply:
I'd actually prefer having these on separate lines for clarity.